### ISSUE TITLE
Secondary Dimension

### ISSUE THREAD

**Issue by mindbreaker** [matomo-org/matomo#16495]
In the current Matomo Version i miss the feature of an secondary dimension.

It is a very useful feature in web analysis. In Google Analytics you can quickly and easily add a secondary dimension to any table view and thus combine the two dimensions. https://support.google.com/analytics/answer/6175970?hl=en

**Comment by tsteur** [comment]
Thanks for creating the issue @mindbreaker I thought we already had an issue about this but I can't find it. Note: The current workaround is to use [Custom Reports](https://plugins.matomo.org/CustomReports) which lets you combine up to three dimensions. Of course it's not the very same as the secondary dimension feature in GA where you simply select this another dimension on any report.

**Comment by SangeethaKannas** [comment]
HI @mindbreaker 

Can I work on this issue?

thanks,
Sangeethakanna.

**Comment by Findus23** [comment]
@SangeethaKannas Feel free to contribute any fixes you have for this. But keep in mind that this is not the simplest issue in Matomo and adding secondary dimensions might mean changing many core parts of Matomo (or maybe not, this is just a rought guess by me).

Feel free to also take a look at the list of beginner-friendly issues: https://github.com/matomo-org/matomo/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+wanted%22

**Comment by SangeethaKannas** [comment]
Hi @Findus23  Thanks for your response. Definitely, I will take a look at beginner issues.

**Comment by frinsen** [comment]
+1

Would love to see this feature - we just moved from GA to Matomo.

**Comment by atom-box** [comment]
A user writes:

_The feature I am missing the most in Matomo is being able to
sort and filter any view by anything. I would like to filter all visits by a specific browser. Or see all visitor actions by a certain source. Or... Maybe this is possible somehow but this is much easier in GA. Sort and filter any view by anything and save this as a new view. This is missing._

_I like right now the second dimension on
each view in GA. See attached screenshot. I know Matomo has segments but I
have to create a segment each time i would like to filter all my
referrers by a special browser or something like that._

![image](https://user-images.githubusercontent.com/22874146/157778641-314a58c4-e742-4bcf-b1a8-2a15ca0df1cc.png)

**Comment by atom-box** [comment]
A user writes:

_In Google Analytics there is the possibility to display a second dimension for certain reports. An example: If I call up the entry pages report, I have the option of showing Google Analytics as a two-dimensional source/medium and can see for each entry page through which channels visits came to the page. Of course there are also other applications, e.g. in acquisition it would be interesting to be able to display the device type as a second dimension.
The example from the screenshot is different: the second dimension Medium is displayed here in the report landingpages._
![image](https://user-images.githubusercontent.com/22874146/157778909-ff4386d9-9fa1-4321-bceb-2455f4545f64.png)

**Comment by atom-box** [comment]
We had a couple requests for this today. Here is one:

" I read somewhere that Matomo doesn't support secondary dimensions the same way Google Analytics does."

**Comment by atom-box** [comment]
Email from a customer:

"In the demo, I do not see what is called « Secondary Dimension » in Google Analytics ( = the possibility of cross-referencing several types of KPI in the same dimension). You confirm me that there is not such feasibility in Matomo ?"

**Comment by serviziperinternet** [comment]
I fully agree this missing function in matomo is the weak point. When they add the secondary dimension in all reports I will definitely use it instead of analytics

**Comment by mindbreaker** [comment]
Yes that's right. matomo currently has no secondary dimension like Piwik Pro for example. What you can do is use the Custom Reports plugin to make your own reports.

**Comment by atom-box** [comment]
A user suggested today:

> Matomo analytics:
> - We are really missing the option to customize the standard reports. i.e. adding secondary dimensions, this would be really handy.

**Comment by atom-box** [comment]
_A user suggested:_

**Question**:
What is the feature you are missing the most in Matomo and why?
**Answer**:
Evaluation by second dimension, similar to Google Analytics

**Comment by 9joshua** [comment]
Received another customer request for this.

**Comment by 9joshua** [comment]
Another request for this from a Cloud customer.

**Comment by goochj03** [comment]
An potential cloud customer wants to see this implemented "_What I really miss is the option to easily add secondary dimensions to any dimension. This, is the main reason I have not switched to Matomo yet_"

**Comment by rcontiuk** [comment]
Another user feedback, "we can't analyze it as thoroughly as we would like to" without the second dimension.

**Comment by LauraAndro** [comment]
Another request for this from a Cloud customer.
